src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionString.cs

```diff
@@ -604,6 +610,28 @@ internal System.Data.SqlClient.ApplicationIntent ConvertValueToApplicationIntent
             }
             // ArgumentException and other types are raised as is (no wrapping)
         }
+        internal System.Data.SqlClient.PoolBlockingPeriod ConvertValueToPoolBlockingPeriod()
+        {
+            string value;
+            TryGetParsetableValue(KEY.PoolBlockingPeriod, out value); 
+            if (value == null)
+            {
+                return DEFAULT.PoolBlockingPeriod;
+            }
+
+            try
+            {
+                return DbConnectionStringBuilderUtil.ConvertToPoolBlockingPeriod(KEY.PoolBlockingPeriod, value);
+            }
+            catch (FormatException e)
+            {
+                throw ADP.InvalidConnectionOptionValue(KEY.PoolBlockingPeriod, e);
+            }
+            catch (OverflowException e)
+            {
+                throw ADP.InvalidConnectionOptionValue(KEY.PoolBlockingPeriod, e);
+            }
+        }
```

Comment:
Nit: Add a new line after method ends.

Reply:
Since the catch blocks are identical, this could also be:
```C#
catch (Exception e) where (e is FormatException || e is OverflowException)
{
    throw ADP.InvalidConnectionOptionValue(KEY.PoolBlockingPeriod, e);
}
```
rather than duplicating.